### DISCUSSION TITLE
Scaling an axis of a mesh

### DISCUSSION THREAD

**Discussion by bhawesh-bks** [pyvista/pyvista#4260] [Q&A] [answered]
So i have plotted a 3D plot using the structured grid and the dimension of the plot is 101*101*21. The range of x axis is 85.5 to 86 and the range of y axis is 22.75-23.25. The range of z axis is -25 to -8250. After plotting iam only getting a single line and i think its because of the difference in the range values of x,y and z. So can you please suggest a way to overcome this issue?
![Screenshot (13)](https://user-images.githubusercontent.com/97894832/231130847-02693028-a642-4450-b5f4-2c2c3eb36f8f.png)

**Comment by banesullivan** [comment, marked as answer]
> and i think its because of the difference in the range values of x,y and z. So can you please suggest a way to overcome this issue?

That is exactly the issue. Try scaling the data itself:

```py
mesh.points[:, -1] /= 100.0
```

or scaling the scene after plotting: https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.set_scale.html#set-scale

```py
pl = pv.Plotter()
pl.add_mesh(mesh)
pl.set_scale(zscale=0.001)
pl.show()
```
